app/src/test/java/org/simple/clinic/patient/recent/RecentPatientsViewControllerTest.kt

```diff
@@ -0,0 +1,84 @@
+package org.simple.clinic.patient.recent
+
+import com.nhaarman.mockito_kotlin.mock
+import com.nhaarman.mockito_kotlin.verify
+import com.nhaarman.mockito_kotlin.whenever
+import io.reactivex.Observable
+import io.reactivex.plugins.RxJavaPlugins
+import io.reactivex.schedulers.Schedulers
+import io.reactivex.subjects.PublishSubject
+import io.reactivex.subjects.Subject
+import junitparams.JUnitParamsRunner
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.simple.clinic.facility.FacilityRepository
+import org.simple.clinic.patient.Age
+import org.simple.clinic.patient.Gender.TRANSGENDER
+import org.simple.clinic.patient.PatientMocker
+import org.simple.clinic.patient.PatientRepository
+import org.simple.clinic.patient.recent.RecentPatient.LastBp
+import org.simple.clinic.user.UserSession
+import org.simple.clinic.util.RxErrorsRule
+import org.simple.clinic.widgets.ScreenCreated
+import org.simple.clinic.widgets.UiEvent
+import org.threeten.bp.Instant
+import org.threeten.bp.LocalDate
+import java.util.UUID
+
+@RunWith(JUnitParamsRunner::class)
+class RecentPatientsViewControllerTest {
+
+  @get:Rule
+  val rxErrorsRule = RxErrorsRule()
+
+  private val screen: RecentPatientsView = mock()
+  private val userSession: UserSession = mock()
+  private val patientRepository: PatientRepository = mock()
+  private val facilityRepository: FacilityRepository = mock()
+
+  private val uiEvents: Subject<UiEvent> = PublishSubject.create()
+  private val loggedInUser = PatientMocker.loggedInUser()
+  private val facilityUuid = UUID.randomUUID()
+
+  @Before
+  fun setUp() {
+    // This is needed because we manually subscribe to the refresh user status
+    // operation on the IO thread, which was causing flakiness in this test.
+    RxJavaPlugins.setIoSchedulerHandler { Schedulers.trampoline() }
+
+    val controller = RecentPatientsViewController(
+        userSession = userSession,
+        patientRepository = patientRepository,
+        facilityRepository = facilityRepository
+    )
+
+    uiEvents
+        .compose(controller)
+        .subscribe { uiChange -> uiChange(screen) }
+  }
+
+  @Test
+  fun `when screen opens then fetch and set recent patients`() {
+    whenever(userSession.requireLoggedInUser()).thenReturn(Observable.just(loggedInUser))
+    whenever(facilityRepository.currentFacilityUuid(loggedInUser)).thenReturn(facilityUuid)
+    whenever(patientRepository.recentPatients(facilityUuid, 10)).thenReturn(Observable.just(listOf(
```

Comment:
Can you add a few more patients in this list with empty values as well?
Say `lastBp` is `null`.